src/rpc/core_rpc_server.cpp

```diff
@@ -205,14 +205,16 @@ namespace cryptonote
       }
       size_t txidx = 0;
       ntxes += bd.second.size();
-      for(const auto& t: bd.second)
+      for (std::list<cryptonote::blobdata>::iterator i = bd.second.begin(); i != bd.second.end(); ++i)
       {
+        unpruned_size += i->size();
         if (req.prune)
-          res.blocks.back().txs.push_back(get_pruned_tx_blob(t));
+          res.blocks.back().txs.push_back(get_pruned_tx_blob(std::move(*i)));
         else
-          res.blocks.back().txs.push_back(t);
+          res.blocks.back().txs.push_back(std::move(*i));
+       i->clear();
```

Comment:
Given that the one move above does not actually occur, I would add `i->shrink_to_fit();`. Clear resets the "size" to 0 but often does not mess with the internal allocated buffer.